pilot/pkg/networking/grpcgen/lds.go

```diff
@@ -281,6 +285,28 @@ func buildOutboundListeners(node *model.Proxy, push *model.PushContext, filter l
 				if !match.includesPort(sPort) {
 					continue
 				}
+				filters := supportedFilters
+				sessionCookie := sv.Attributes.Labels["x-session-cookie"]
+				if sessionCookie != "" {
+					filters = append(filters, &hcm.HttpFilter{
+						Name: "envoy.filters.http.stateful_session", // TODO: wellknown.
+						ConfigType: &hcm.HttpFilter_TypedConfig{
+							TypedConfig: protoconv.MessageToAny(&stateful_sessionv3.StatefulSession{
+								SessionState: &core.TypedExtensionConfig{
+									Name: "envoy.http.stateful_session.cookie",
+									TypedConfig: protoconv.MessageToAny(&cookiev3.CookieBasedSessionState{
```

Comment:
Regarding `stateful_sessionv3.StatefulSession` and `cookiev3.CookieBasedSessionState` : I don't know how they translate to `@type`  `type.googleapis.com/envoy.extensions.filters.http.stateful_session.v3.StatefulSession` and `type.googleapis.com/envoy.extensions.http.stateful_session.cookie.v3.CookieBasedSessionState` strings respectively. May be it's all right but just pointing it out

Reply:
AFAIK MessageToAny will set the type automatically, at least they do in other cases.